test/integration/timeout/timeout_stack.go

```diff
@@ -0,0 +1,666 @@
+	var deletionErrors []error
+	By("delete all resources for services", func() {
+		By(fmt.Sprintf("Delete backend VirtualService"), func() {
+			if err := f.K8sClient.Delete(ctx, s.BackEndVS); err != nil {
+				f.Logger.Error("failed to delete VirtualService",
+					zap.String("namespace", s.BackEndVS.Namespace),
+					zap.String("name", s.BackEndVS.Name),
+					zap.Error(err),
+				)
+				deletionErrors = append(deletionErrors, err)
+			}
+		})
+
+		By(fmt.Sprintf("Delete backend VirtualRouter"), func() {
+			if err := f.K8sClient.Delete(ctx, s.BackEndVR); err != nil {
+				f.Logger.Error("failed to delete VirtualRouter",
+					zap.String("namespace", s.BackEndVR.Namespace),
+					zap.String("name", s.BackEndVR.Name),
+					zap.Error(err),
+				)
+				deletionErrors = append(deletionErrors, err)
+			}
+		})
+
+		By("Wait for backend VirtualService to be deleted", func() {
+			if err := f.VSManager.WaitUntilVirtualServiceDeleted(ctx, s.BackEndVS); err != nil {
+				f.Logger.Error("failed to delete backend VirtualService", zap.Error(err))
+				return
+			}
+		})
+		By("Wait for backend VirtualRouter to be deleted", func() {
+			if err := f.VRManager.WaitUntilVirtualRouterDeleted(ctx, s.BackEndVR); err != nil {
+				f.Logger.Error("failed to delete backend VirtualRouter", zap.Error(err))
+				return
+			}
+		})
+	})
+	return deletionErrors
+}
+
+func (s *TimeoutStack) assignBackendVSToFrontEndVN(ctx context.Context, f *framework.Framework) {
+	By("granting VirtualNodes backend access", func() {
+		By(fmt.Sprintf("assigning backend VS to frontend VN"), func() {
+			var vnBackends []appmesh.Backend
+			vs := s.BackEndVS
+			vnBackends = append(vnBackends, appmesh.Backend{
+				VirtualService: appmesh.VirtualServiceBackend{
+					VirtualServiceRef: &appmesh.VirtualServiceReference{
+						Namespace: aws.String(vs.Namespace),
+						Name:      vs.Name,
+					},
+				},
+			})
+
+			vnNew := s.FrontEndVN.DeepCopy()
+			vnNew.Spec.Backends = vnBackends
+			err := f.K8sClient.Patch(ctx, vnNew, client.MergeFrom(s.FrontEndVN))
+			Expect(err).NotTo(HaveOccurred())
+			s.FrontEndVN = vnNew
+		})
+	})
+}
+
+func (s *TimeoutStack) revokeVirtualNodeBackendAccessForTimeoutStack(ctx context.Context, f *framework.Framework) []error {
+	var deletionErrors []error
+	By("revoking VirtualNodes backend access", func() {
+		By(fmt.Sprintf("revoking Frontend VirtualNode access to backend"), func() {
+			vnNew := s.FrontEndVN.DeepCopy()
+			vnNew.Spec.Backends = nil
+
+			err := f.K8sClient.Patch(ctx, vnNew, client.MergeFrom(s.FrontEndVN))
+			if err != nil {
+				f.Logger.Error("failed to revoke VirtualNode backend access",
+					zap.String("namespace", s.FrontEndVN.Namespace),
+					zap.String("name", s.FrontEndVN.Name),
+					zap.Error(err),
+				)
+				deletionErrors = append(deletionErrors, err)
+			}
+		})
+	})
+	return deletionErrors
+}
+
+func (s *TimeoutStack) checkDefaultBackendRouteTimesout(ctx context.Context, f *framework.Framework,
```

Comment:
`checkDefaultBackendRouteTimesout` and `checkBehaviorOfRouteConfiguredWithTimeout` are very similar except for the one line of error message check. You could merge them into one function and have a check based on a condition